awsiot/iotshadow.py

```diff
@@ -1295,15 +1303,22 @@ class ShadowState(awsiot.ModeledClass):
 
     Attributes:
         desired (typing.Dict[str, typing.Any]): The desired shadow state (from external services and devices).
+        desired_none_is_valid (bool): Set to true to allow desired to be None, clearing the data if sent.
+
         reported (typing.Dict[str, typing.Any]): The (last) reported shadow state from the device.
+        reported_none_is_valid (bool): Set to true to allow reported to be None, clearing the data if sent.
+
     """
 
-    __slots__ = ['desired', 'reported']
+    __slots__ = ['desired', 'desired_none_is_valid', 'reported', 'reported_none_is_valid']
 
     def __init__(self, *args, **kwargs):
         self.desired = kwargs.get('desired')
         self.reported = kwargs.get('reported')
 
+        self.desired_none_is_valid = kwargs.get('desired_none_is_valid')
+        self.reported_none_is_valid = kwargs.get('reported_none_is_valid')
```

Comment:
you have the new  `xyz_none_is_valid` values defaulting to `None` but we should have them defaulting to `False`

Reply:
Fixed!

Reply:
misunderstanding.

I was saying that `kwargs.get('desired_none_is_valid')` will return `None` by default

but you can do: `kwargs.get('desired_none_is_valid', False)` to get `False` instead